### ISSUE TITLE
Uncaught ReferenceError: __extends is not defined

### ISSUE THREAD

**Issue by MeindertRomkes** [akveo/ng2-smart-table#16]
Hello, 

For some reason I get this error mentioned in the title. it has something to with my LocalDataSource but I don't know how en why. I am enthusiastic about this data-table and I would like to use it. 

Kind regards,
Meindert

**Comment by nnixaa** [comment]
Hey @MeindertRomkes, would you mind sharing a code example or plunkr to take a look at?

**Comment by re5pawn** [comment]
I have the same problem! I just used code from [Minimal Setup Example](https://github.com/akveo/ng2-smart-table#minimal-setup-example) and bundled it by webpack.

* ng2-smart-table: 0.2.2
* angular2: rc.5
* webpack: `{ test: /\.ts$/, loader: 'awesome-typescript!angular2-template' }`

**Comment by KingsDevelopment** [comment]
me and @MeindertRomkes are using the angular-cli version of our project, mainly everything is managed by that + webpack.

**Comment by nnixaa** [comment]
guys, thanks for reporting, I would need a bit more information about the environment, webpack, typescript etc version, webpack configurations to investigate the issue. Also version of the module you use.

Generally you could take a look at our demo webpack/package configurations to compare:
https://github.com/akveo/ng2-smart-table/blob/master/config/webpack.common.js#L138
https://github.com/akveo/ng2-smart-table/blob/master/package.json

**Comment by KingsDevelopment** [comment]
webpack version: webpack 2.1.0-beta.18 (Angular CLI)
webpack configuration is handled by angular cli, (more information)[https://github.com/angular/angular-cli]

typescript version 2.0.0

module version: latest of npm.


Environment: Ubuntu docker container.

**Comment by re5pawn** [comment]
* ng2-smart-table: 0.2.2
* angular2: rc.5
* webpack: 1.13.2 - { test: /\.ts$/, loader: 'awesome-typescript!angular2-template' }
* typescript: 1.8.10
* windows 10 pro 64-bit

**Comment by nnixaa** [comment]
all right, could you also provide a complete error output including stack trace?

**Comment by re5pawn** [comment]
Here you are:
```js
main.bundle.js:24655 Uncaught ReferenceError: __extends is not defined

(anonymous function) @ main.bundle.js:24655
(anonymous function) @ main.bundle.js:24880
__webpack_require__ @ bootstrap aa341a8…:50
(anonymous function) @ main.bundle.js:25095
__webpack_require__ @ bootstrap aa341a8…:50
(anonymous function) @ main.bundle.js:24609
__webpack_require__ @ bootstrap aa341a8…:50
(anonymous function) @ main.bundle.js:27734
__webpack_require__ @ bootstrap aa341a8…:50
(anonymous function) @ main.bundle.js:25683
__webpack_require__ @ bootstrap aa341a8…:50
(anonymous function) @ main.bundle.js:7
__webpack_require__ @ bootstrap aa341a8…:50
webpackJsonpCallback @ bootstrap aa341a8…:21
(anonymous function) @ main.bundle.js:1
```

**Comment by nnixaa** [comment]
okay, love the stacktraces like this 😞 , would you guys be able to provide a minimal test repo using your env, please? I'm struggling to reproduce the issue with no luck.

**Comment by trf19** [comment]
We are facing the same issue trying to get ng2-smart-table to work within our project. 

ng2-smart-table: 0.2.3
angular2: rc.6
webpack: 1.13.2
typescript: 2.1.0-dev.20160823

**stack trace:**
local.data-source.js:8Uncaught ReferenceError: __extends is not defined(anonymous function) 
@ local.data-source.js:8(anonymous function)
@ local.data-source.js:232n 
@ bootstrap fe736e7…:50(anonymous function) 
@ ng2-smart-table.component.js:6n 
@ bootstrap fe736e7…:50(anonymous function) 
@ ng2-smart-table.directives.js:2n 
@ bootstrap fe736e7…:50(anonymous function) 
@ ng2-smart-table.js:5n 
@ bootstrap fe736e7…:50(anonymous function) 
@ pages.module.ts:19n 
@ bootstrap fe736e7…:50(anonymous function) 
@ app.module.ts:13n 
@ bootstrap fe736e7…:50(anonymous function) 
@ main.ts:3n 
@ bootstrap fe736e7…:50(anonymous function) 
@ app.fe736e7….js:6n 
@ bootstrap fe736e7…:50window.webpackJsonp 
@ bootstrap fe736e7…:21(anonymous function) 
@ app.fe736e7….js:1

Happens as soon as I try adding Ng2SmartTableModule into the "imports" of my NgModule. Unfortunately at this time I cannot provide a repo to reproduce the issue.

**Comment by Tadwork** [comment]
related to #26 , the fix is to add the typescript helpers globally for now , or to compile it with `noEmitHelpers:false`

**Comment by bvlangen** [comment]
Issue is also reported under issue #16

**Comment by MeindertRomkes** [comment]
Thanks, the fix worked for me.

**Comment by sambanthar** [comment]
where i need to add noEmitHelpers:false in node module ng2-smart-table or in my project root tsconfig please help us

**Comment by nnixaa** [comment]
fixed in 0.4.0-3
